src/utilcode/prettyprintsig.cpp

```diff
@@ -866,6 +839,67 @@ static HRESULT PrettyPrintTypeA(
 #pragma warning(pop)
 #endif
 
+// pretty prints the class 'type' to the buffer 'out'
+static HRESULT PrettyPrintClass(
+    PCCOR_SIGNATURE     &typePtr,   // type to convert
+    CQuickBytes         *out,       // where to put the pretty printed string
+    IMDInternalImport   *pIMDI)     // ptr to IMDInternal class with ComSig
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        INJECT_FAULT(return E_OUTOFMEMORY;);
+    }
+    CONTRACTL_END
+
+    mdToken     tk;
+    const CHAR  *str;   // type's token.
+    LPCUTF8     pNS;    // type's namespace.
+    LPCUTF8     pN;     // type's name.
+    HRESULT     hr;     // result
+
+    typePtr += CorSigUncompressToken(typePtr, &tk);
```

Comment:
It would be nice to do `IfFailGo(CorSigUncompressToken_EndPtr(typePtr, typeEnd , &tk));` here to make sure we are not reading beyond the end of the buffer. (This bug was there even before your change.)

Reply:
Done.